### PR TITLE
[TH2-4566] Owasp dependency check build step

### PR DESCRIPTION
**Pull request by isengrims**

### CODE REVIEW

**Review by OptimumCode — CHANGES_REQUESTED** [review]

**Review by OptimumCode — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -128,4 +128,15 @@ jobs:
           push: true
           context: .
           tags: ghcr.io/${{ github.repository }}:${{ needs.metadata.outputs.release_version }}
-          labels: com.exactpro.th2.${{ steps.meta.outputs.REPOSITORY_NAME }}=${{ needs.metadata.outputs.release_version }}
\ No newline at end of file
+          labels: com.exactpro.th2.${{ steps.meta.outputs.REPOSITORY_NAME }}=${{ needs.metadata.outputs.release_version }}
+  owasp-scan-job:
+    runs-on: ubuntu-20.04
```
Comment (OptimumOpium) [inline .github/workflows/build.yml:133]: You can reuse job from our corporate workflows. There were changes to support multiprojects

File: build.gradle [hunk 2]
```diff
@@ -100,4 +102,12 @@ nexusPublishing {
...
     }
+}
+
+allprojects {
+    apply plugin: 'org.owasp.dependencycheck'
+
+    dependencyCheck {
+        formats=['SARIF']
+    }
 }
```
Comment (OptimumOpium) [inline build.gradle:113]: You need to apply the plugin and configure the task only for the root project.

**Review by OptimumCode — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -10,122 +10,78 @@ on:
...
+        grpc: ${{ steps.filter.outputs.grpc }}
+      steps:
+        - uses: dorny/paths-filter@v2
+          id: filter
+          with:
+            filters: |
+              app:
+                - 'gradle.properties'
+              grpc:
+                - 'grpc/gradle.properties'
```
Comment (OptimumCode) [inline .github/workflows/build.yml]: I don't think this  work when you use `uses` directive.
Probably you wanted this:
```suggestion
  changes:
      name: Collect changed files
      runs-on: ubuntu-20.04
      outputs:
        app: ${{ steps.filter.outputs.app }}
        grpc: ${{ steps.filter.outputs.grpc }}
      steps:
        - uses: dorny/paths-filter@v2
          id: filter
          with:
            filters: |
              app:
                - 'gradle.properties'
              grpc:
                - 'grpc/gradle.properties'
```

File: build.gradle [hunk 4]
```diff
@@ -28,8 +29,27 @@ ext {
...
+        nodeEnabled = false
+    }
+}
+
+group = 'com.exactpro.th2'
+version = release_version
+String suffix = property('version_suffix').toString()
+if (!suffix.isEmpty()) {
+    version = "$version-$suffix"
+}
```
Comment (OptimumCode) [inline build.gradle]: Please, move this into `buildSrc/src/main/groovy/com.exactpro.th2.common-convensions.gradle`

File: grpc/build.gradle [hunk 5]
```diff
@@ -2,18 +2,25 @@ import static org.gradle.api.file.DuplicatesStrategy.*
...
+    serviceGeneratorVersion = '3.3.0'
+    genBaseDir = file("${buildDir}/generated/source/proto") // from protobuf plugin
+}
+
+group = 'com.exactpro.th2'
+version = release_version
+String suffix = property('version_suffix').toString()
+if (!suffix.isEmpty()) {
+    version = "$version-$suffix"
 }
```
Comment (OptimumCode) [inline grpc/build.gradle]: Will be applied from `buildSrc/src/main/groovy/com.exactpro.th2.common-convensions.gradle`

File: grpc/build.gradle [hunk 6]
```diff
@@ -28,13 +35,11 @@ dependencies {
 sourceSets {
     main {
         proto {
-            srcDirs 'src/main/proto'
-        }
-        java {
-            srcDirs 'src/gen/main/java', 'src/gen/main/grpc', 'src/gen/main/services/java/src'
+            srcDirs 'proto'
```
Comment (OptimumCode) [inline grpc/build.gradle:31]: Have you moved proto files into that directory?
You could look at this repo as an example: https://github.com/th2-net/th2-read-db/tree/dev/grpc

**Review by OptimumCode — CHANGES_REQUESTED** [review]

File: grpc/build.gradle [hunk 7]
```diff
@@ -77,5 +75,5 @@ tasks {
 }
 
 clean {
-    delete genDir
+    delete genBaseDir
 }
```
Comment (OptimumCode) [inline grpc/build.gradle:79]: Can be removed because the directory is located in `build` folder

File: grpc/gradle.properties [hunk 8]
```diff
@@ -1,3 +1,3 @@
 kotlin.code.style=official
-
+version_suffix=
```
Comment (OptimumCode) [inline grpc/gradle.properties]: No need to add this. It will be taken from root gradle.properties

File: grpc/build.gradle [hunk 9]
```diff
@@ -30,11 +30,9 @@ sourceSets {
         proto {
             srcDirs 'src/main/proto'
```
Comment (OptimumCode) [inline grpc/build.gradle]: Please, correct the project's structure as it was done in https://github.com/th2-net/th2-read-db/tree/dev/grpc

File: grpc/proto/th2_grpc_lw_data_provider/lw_data_provider.proto [hunk 10]
```diff
@@ -1,223 +1,223 @@
...
-  google.protobuf.Timestamp ended = 4;
-  google.protobuf.Timestamp updated = 5;
-  google.protobuf.Timestamp removed = 6;
-}
-
-message PageId {
-  BookId book_id = 1;
-  string name = 2;
+/*
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
```
Comment (OptimumCode) [inline grpc/proto/th2_grpc_lw_data_provider/lw_data_provider.proto]: Please, update copyright

File: grpc/proto/th2_grpc_lw_data_provider/lw_data_provider.proto [hunk 11]
```diff
@@ -1,223 +1,223 @@
...
-  google.protobuf.Timestamp started = 3;
-  google.protobuf.Timestamp ended = 4;
-  google.protobuf.Timestamp updated = 5;
-  google.protobuf.Timestamp removed = 6;
-}
-
-message PageId {
-  BookId book_id = 1;
-  string name = 2;
+/*
```
Comment (OptimumCode) [inline grpc/proto/th2_grpc_lw_data_provider/lw_data_provider.proto:1]: Have you changed the formatting? Why the file is changed completely?

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**